### PR TITLE
LicenseStore migration to return plain errors

### PR DESCRIPTION
**Pull request by rvillablanca**
#### Summary

This PR migrates the LicenseStore to return plain errors

#### Ticket Link
 
Fixes https://github.com/mattermost/mattermost-server/issues/14816

### CODE REVIEW

File: store/sqlstore/license_store.go [hunk 1]
```diff
@@ -66,13 +68,13 @@ func (ls SqlLicenseStore) Save(license *model.LicenseRecord) (*model.LicenseReco
...
-func (ls SqlLicenseStore) Get(id string) (*model.LicenseRecord, *model.AppError) {
+func (ls SqlLicenseStore) Get(id string) (*model.LicenseRecord, error) {
 	obj, err := ls.GetReplica().Get(model.LicenseRecord{}, id)
 	if err != nil {
-		return nil, model.NewAppError("SqlLicenseStore.Get", "store.sql_license.get.app_error", nil, "license_id="+id+", "+err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrapf(err, "failed to get License with licenseId=%s", id)
 	}
 	if obj == nil {
-		return nil, model.NewAppError("SqlLicenseStore.Get", "store.sql_license.get.missing.app_error", nil, "license_id="+id, http.StatusNotFound)
+		return nil, store.NewErrNotFound("License", fmt.Sprintf("licenceId=%s", id))
```
Comment (agnivade) [inline store/sqlstore/license_store.go:77]: Is it necessary to pass `licenseId=`? The error already assumes that it's an id, and prints `id: `.

File: store/sqlstore/license_store.go [hunk 2]
```diff
@@ -50,13 +52,13 @@ func (ls SqlLicenseStore) Save(license *model.LicenseRecord) (*model.LicenseReco
 		Where(sq.Eq{"Id": license.Id})
 	queryString, args, err := query.ToSql()
 	if err != nil {
-		return nil, model.NewAppError("SqlLicenseStore.Save", "store.sql.build_query.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrap(err, "failed to build query to get Licenses")
```
Comment (agnivade) [inline store/sqlstore/license_store.go:55]: I guess this needs to be "license_tosql" ?

File: app/license.go [hunk 3]
```diff
@@ -74,10 +74,10 @@ func (s *Server) SaveLicense(licenseBytes []byte) (*model.License, *model.AppErr
 	record.Id = license.Id
 	record.Bytes = string(licenseBytes)
 
-	_, err = s.Store.License().Save(record)
-	if err != nil {
+	_, nErr := s.Store.License().Save(record)
+	if nErr != nil {
 		s.RemoveLicense()
-		return nil, model.NewAppError("addLicense", "api.license.add_license.save.app_error", nil, "err="+err.Error(), http.StatusInternalServerError)
+		return nil, model.NewAppError("addLicense", "api.license.add_license.save.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (agnivade) [inline app/license.go:80]: `Save` also returns AppError. Don't we need to check that here and return the apperror directly?

File: store/sqlstore/license_store.go (hunk 1 above)
Comment (rvillablanca) [reply store/sqlstore/license_store.go:77]: you are right, I forgot that

File: store/sqlstore/license_store.go (hunk 2 above)
Comment (rvillablanca) [reply store/sqlstore/license_store.go:55]: Ok

**Comment by agnivade** [comment]
@ogi-m - This needs some testing around saving and loading of licenses. Thanks.

**Pull request merged**